src/Worker/Core/Shims/TaskOrchestrationContextWrapper.cs

```diff
@@ -38,9 +38,31 @@ public TaskOrchestrationContextWrapper(
         OrchestrationContext innerContext,
         OrchestrationInvocationContext invocationContext,
         object? deserializedInput)
+        : this(innerContext, invocationContext, deserializedInput, new Dictionary<string, object?>())
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="TaskOrchestrationContextWrapper"/> class.
+    /// </summary>
+    /// <param name="innerContext">The inner orchestration context.</param>
+    /// <param name="invocationContext">The invocation context.</param>
+    /// <param name="deserializedInput">The deserialized input.</param>
+    /// <param name="properties">The configuration for context.</param>
+    public TaskOrchestrationContextWrapper(
+        OrchestrationContext innerContext,
+        OrchestrationInvocationContext invocationContext,
+        object? deserializedInput,
+        IEnumerable<KeyValuePair<string, object?>> properties)
     {
         this.innerContext = Check.NotNull(innerContext);
         this.invocationContext = Check.NotNull(invocationContext);
+        if (properties is null)
+        {
+            throw new ArgumentNullException(nameof(properties));
+        }
+
+        this.Properties = properties.ToDictionary(pair => pair.Key, pair => pair.Value);
```

Comment:
Converting from `IEnumerable<KVP>` to a dictionary is dangerous because `IEnumerable<KVP>` allows duplicate keys whereas dictionary does not. We should change the `properties` parameter to be a dictionary instead of `IEnumerable<KVP>` to avoid these kinds of problems.

Reply:
It is safe if you use a `foreach` loop and set the key each time, so last key wins.

Reply:
Thanks, Chris and Jacob. I updated this to be Dictionary. Because to write a foreach loop, I have to either grant Properties `private set`, or adding a new field to store this and then pass the value to Properties. Just feel like using Dictionary directly here seems like a more straightforward and cleaner way in this case.

Reply:
I am okay with going back to dictionary, but I don't see why you need a `private set`. The following is valid without any setter on `Properties`:

``` CSharp
this.Properties = new();
foreach ((string key, object? value) in properties)
{
    this.Properties[key] = value;
}
```